### DISCUSSION TITLE
Will Prisma return all the nested children of a 1-to-many relationship on self

### DISCUSSION THREAD

**Discussion by gj1118** [prisma/prisma#16817] [Q&A]
Hi, 
I am currently using Prisma 3.15.2 ( have yet to upgrade to the latest version of Prisma 4). I am having a one to many relationship with a model. Please see the schema definition below 

```
  parent           Int?
  parentAction     Action?         @relation("ParentAction", fields: [parent], references: [id])
  childActions     Action[]        @relation("ParentAction")
```

A parent action can have multiple child actions and childActions can have further child actions. What I am trying to ask is → will prisma return me the entire nested list or only the first level. I am asking this because in my testing, I am seeing that its returning me the first level only. Perhaps I am doing something wrong here?  

Can somebody please confirm as and when its possible, please ?  

Thanks once again for all the help. 

Much appreciated. 

Gagan

**Comment by nurul3101** [comment]
Hello @gj1118 👋 

Can you share the query that you are invoking? Prisma will not automatically return all the nested relation fields, You would need to use [include](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#include) to get all the relation fields which are more than one level deep.

If you can share the query, I can have a look and investigate how you can achieve nested relation field results.

**Comment by gj1118** [comment]
@nurul3101 , thanks for taking time out. What I am asking is not related to the nested fields, but related to getting all the nested data in a one to many relationship on the same model. For example I have a parent A. A has a child B while B has a child C and C has children D,E, F . 

Now when I query for A and its related children, will A return B or will A return B and all its subsequent children and grandchildren?  

Thanks. 

With Regards, 
Gagan

**Reply by nurul3101** [reply]
@gj1118 No, by default Prisma will not return all nested data at all levels. We have a Feature Request for it here
- #3725 

If you can leave a 👍 to the feature request then it would help our product team in prioritising it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
